crates/handlers/src/compat/login.rs

```diff
@@ -193,7 +194,7 @@ pub enum RouteError {
     NoPassword,
 
     #[error("password verification failed")]
-    PasswordVerificationFailed(#[source] anyhow::Error),
+    PasswordVerificationFailed,
```

Comment:
To be consistent with the rest, split this into `PasswordMismatch` and `PasswordVerificationFailed(#[source] anyhow::Error)`, and mark the second as an internal server error like we do for `ProvisionDeviceFailed`

Reply:
Have renamed. Errors raised during verification are currently going to `InternalError`, is it helpful to split that up into a separate variant? I wouldn't have thought so as they're unexpected internal errors at that point.